fil_fungible_token/src/token/mod.rs

```diff
@@ -80,28 +89,64 @@ where
     }
 }
 
-impl<BS> Token<BS>
+impl<BS, MC> Token<BS, MC>
 where
     BS: IpldStore + Clone,
+    MC: MethodCaller,
 {
     /// Mints the specified value of tokens into an account
     ///
     /// The mint amount must be non-negative or the method returns an error
-    pub fn mint(&mut self, initial_holder: ActorID, value: TokenAmount) -> Result<()> {
+    pub fn mint(
+        &mut self,
+        minter: ActorID,
+        initial_holder: ActorID,
+        value: &TokenAmount,
+        data: &[u8],
+    ) -> Result<()> {
         if value.is_negative() {
             return Err(TokenError::InvalidNegative(format!(
                 "mint amount {} cannot be negative",
                 value
             )));
         }
 
+        let old_state = self.state.clone();
```

Comment:
Note: It's slightly sad that this clone followed by the transaction results in a redundant clone, and that we do an actual write to rollback when the old state is often already persisted. The state is small so it doesn't matter _too_ much. 

An alternative pattern would involve keeping track of the CID at load, and reverting by re-loading it. It this presents a snapshot (could avoid all the clones this way, but I'm not sure which is cheaper). It doesn't immediately apply here tho since we don't know whether the prior state has been modified since load.

Reply:
![image](https://user-images.githubusercontent.com/7680266/181158264-e4a88378-c576-40e2-a946-34334de7c3c7.png)

A quick test using the fvm_integration_tests package seems to show that the clone and revert strategy is cheaper than reloading the old state from the blockstore

Since the root level of state is just a BigInt and 2 CIDs, the only real difference is whether we revert those three values from an in-memory copy (should be cheap) or a copy in the blockstore. In fact, the testing I did seems to suggest that there's no gas difference in making that clone (but not sure if there's something wrong with the testing methodology here).

Reply:
Thanks, I'm happy for us to roll with clones then.